api/post.go

```diff
@@ -432,19 +436,27 @@ func getFileInfosForPost(c *Context, w http.ResponseWriter, r *http.Request) {
 func getOpenGraphMetadata(c *Context, w http.ResponseWriter, r *http.Request) {
 	props := model.StringInterfaceFromJson(r.Body)
 
+	ogJSONGeneric, ok := openGraphDataCache.Get(props["url"])
+	if ok {
+		w.Write(ogJSONGeneric.([]byte))
+		return
+	}
+
 	url := ""
-	ok := false
+	ok = false
 	if url, ok = props["url"].(string); len(url) == 0 || !ok {
 		c.SetInvalidParam("getOpenGraphMetadata", "url")
 		return
 	}
 
 	og := app.GetOpenGraphMetadata(url)
 
-	ogJson, err := og.ToJSON()
+	ogJSON, err := og.ToJSON()
 	if err != nil {
 		w.Write([]byte(`{"url": ""}`))
 		return
 	}
-	w.Write(ogJson)
+
+	openGraphDataCache.Add(props["url"], ogJSON)
```

Comment:
Should probably use `AddWithExpiresInSecs()` here so that the cache will expire after 15 minutes or so and catch the any changes on the webpage